json_encoder.go

```diff
@@ -167,6 +165,28 @@ func (enc *jsonEncoder) AddMarshaler(key string, obj LogMarshaler) error {
 	return err
 }
 
+// AddInts adds a string key and integer slice to the encoder's fields. The key
+// is JSON-escaped.
+func (enc *jsonEncoder) AddInts(key string, vals []int) {
+	enc.beginArray(key)
+	for i := range vals {
+		enc.addArraySep(i)
+		enc.addInt64(int64(vals[i]))
+	}
+	enc.endArray()
+}
+
+// AddStrings adds a string key and string slice to the encoder's fields. The key
+// and the slice elements are JSON-escaped.
+func (enc *jsonEncoder) AddStrings(key string, vals []string) {
+	enc.beginArray(key)
+	for i := range vals {
+		enc.addArraySep(i)
+		enc.addAndQuoteString(vals[i])
```

Comment:
Is there a benefit of using vals[i] over val (from for i, val := range vals)?  The latter is a bit more error-proof.